File: hw/CEDENOR/U3/HW30Observer&Strategy/Java/ObserverPatttern/src/main/java/ec/edu/espe/observerpattern/view/Application.java

package ec.edu.espe.observerpattern.view;
import ec.edu.espe.observerpattern.controller.*;
/**
 *
 * @author Nahomi Cedeño, codecreators, DCCO - ESPE
 */

public class Application {
    public static void main(String[] args) {
      
        Investor s = new Investor("Sorros");
        Investor b = new Investor("Berkshire");

        IBM ibm = new IBM("IBM", 120.0);

        ibm.addObserver(s);
        ibm.addObserver(b);

        ibm.setPrice(120.10);
        ibm.setPrice(121.00);
        ibm.setPrice(120.50);
        ibm.setPrice(120.75);

        ibm.setSymbol("IBMTEST");
    }
}

File: hw/CEDENOR/U3/HW30Observer&Strategy/Python/ObserverPython/view/application.py
import sys
import os
# Nahomi Cedeño, codecreators, DCCO - ESPE

sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
from controller.IBM import IBM
from controller.Investor import Investor

def main():
    
    s = Investor("Sorros")
    b = Investor("Berkshire")

    ibm = IBM("IBM", 120.0)

    ibm.add_observer(s)
    ibm.add_observer(b)

    ibm.price = 120.10
    ibm.price = 121.00
    ibm.price = 120.50
    ibm.price = 120.75

    ibm.symbol = "IBMTEST"

if __name__ == "__main__":
    main()
